### PR TITLE
[pulsar-broker] cluster-resource use cluster name to fetch cluster-metadata

### PR DESCRIPTION
**Pull request by rdhabalia**
### Motivation
Note: On top of : #9925
Cluster-resources should be able to fetch cluster-metadata and internal-path should be handled by the base-resources. It will make code clean and will be helpful for future abstraction.

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java [hunk 1]
```diff
@@ -390,7 +391,7 @@ public void deleteCluster(
             }
         } catch (Exception e) {
             log.error("[{}] Failed to get cluster usage {}", clientAppId(), cluster, e);
-            throw new RestException(e);
+            throw new RestException(e instanceof ExecutionException ? e.getCause() : e);
```
Comment (eolivelli) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java:394]: What about adding an utility function for this line? I see it written more times

**Review by eolivelli — APPROVED** [review]
LGTM
I left one suggestion PTAL

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java [hunk 2]
```diff
@@ -624,16 +621,18 @@ protected boolean isNamespaceReplicated(NamespaceName namespaceName) {
...
         } catch (Exception e) {
-            log.error("[{}] Failed to get partitioned topic list for namespace {}", clientAppId(),
-                    namespaceName.toString(), e);
-            throw new RestException(e);
+            if (e instanceof ExecutionException && e.getCause() instanceof NotFoundException) {
+                // NoNode means there are no partitioned topics in this domain for this namespace
+            } else {
+                log.error("[{}] Failed to get partitioned topic list for namespace {}", clientAppId(),
+                        namespaceName.toString(), e);
+                throw new RestException(e);
```
Comment (315157973) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java:634]: Would it be better, just a suggestion
`throw new RestException(e instanceof ExecutionException ? e.getCause() : e)`

**Comment by codelipenghui** [comment]
Hi Rajan @rdhabalia, Could you please help resolve the conflicts? so that we can merge it before 2.8.0

**Comment by rdhabalia** [comment]
it seems #11693 again had same changes and that new PR is merged. so, we should close this one.

**Pull request closed without merging**